src/protocol/id/asset.md

```diff
@@ -0,0 +1,8 @@
+# Asset ID
+
+The _asset ID_ (also called _asset hash_) of a asset is computed as
+the [hash](../cryptographic_primitives.md#hashing) of the `CONTRACT_ID` and a 256-bit `SUB_IDENTIFIER`.
```

Comment:
Do we want to support the `None` sub-identifier? In this case, the contract still can mint `AssetId == ContractId`.

Of course, we can't use `None`, but we can use some magic value instead like zeroed sub-identifier or a maximum sub-identifier.

Reply:
I believe the decision was to rip the band-aid off, and break the tie that `AssetId == ContractID`

It'll be more confusing if they're _sometimes_ equal, and sometimes different.

Reply:
If we don't need backward compatibility, then okay=) But it means it is a super breaking change

Reply:
Yep, it's a breaking change, but that's why it's important we get it in soon so teams can adjust